src/main/java/net/snowflake/ingest/streaming/internal/FlushService.java

```diff
@@ -338,33 +338,72 @@ void distributeFlushTasks() {
                 String, ConcurrentHashMap<String, SnowflakeStreamingIngestChannelInternal<T>>>>
         itr = this.channelCache.iterator();
     List<Pair<BlobData<T>, CompletableFuture<BlobMetadata>>> blobs = new ArrayList<>();
+    List<ChannelData<T>> leftoverChannelsDataPerTable = new ArrayList<>();
```

Comment:
Is it possible to forward channel data such that it causes us to go over the size threshold in the next batch?

Reply:
good question, if you see line 353, we will add everything in `leftoverChannelsDataPerTable` to `channelsDataPerTable`, then the same split logic will apply again on `channelsDataPerTable`

Reply:
Got it, so is it possible to continually move data around if there's a high ingest rate? Probably an edge case but something I wanted to call out.

Reply:
I guess the word "split" is not accurate, we basically have an array of chunks and we will see how many blobs we need so the size of each blob stay below the limit, so there is no "data moving", but if the ingest rate is high and we don't have enough threads to build the actual blobs, that will be an issue and we will see a high build latency.

Reply:
Got it - makes sense.